assignments/array-methods.js

```diff
@@ -56,30 +56,56 @@ const runners = [
+// This array will contain just strings.
 let fullNames = [];
+const firstAndLastNames = runners.forEach(person => {
+  console.log(fullNames.push(`${person.first_name} ${person.last_name}`)); 
+});
 console.log(fullNames);
 
 // ==== Challenge 2: Use .map() ====
-// The event director needs to have all the runners' first names in uppercase because the director BECAME DRUNK WITH POWER. Populate an array called `firstNamesAllCaps`. This array will contain just strings.
-let firstNamesAllCaps = [];
+// The event director needs to have all the runners' first names in uppercase because the director BECAME DRUNK WITH POWER. 
+// Populate an array called `firstNamesAllCaps`. This array will contain just strings.
+// let firstNamesAllCaps = [];
+const firstNamesAllCaps = runners.map((person) => {
+  return person.first_name.toUpperCase();
+})
 console.log(firstNamesAllCaps);
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue. We need a filtered version of the runners array, containing only those runners with large sized shirts so they can choose a different size. This will be an array of objects.
-let runnersLargeSizeShirt = [];
+// The large shirts won't be available for the event due to an ordering issue. 
+// We need a filtered version of the runners array, containing only those runners with large sized shirts so they can choose a different size. 
+// This will be an array of objects.
+// let runnersLargeSizeShirt = [];
+const runnersLargeSizeShirt = runners.filter((person) => {
+  return person.shirt_size === "L";
+});
 console.log(runnersLargeSizeShirt);
 
 // ==== Challenge 4: Use .reduce() ====
-// The donations need to be tallied up and reported for tax purposes. Add up all the donations and save the total into a ticketPriceTotal variable.
-let ticketPriceTotal = 0;
+// The donations need to be tallied up and reported for tax purposes. 
+// Add up all the donations and save the total into a ticketPriceTotal variable.
+const ticketPriceTotal = runners.reduce((total, person) => {
+  return total + person.donation;
+},0);
+
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
-// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
+// Now that you have used .forEach(), .map(), .filter(), and .reduce().  
+// I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  
+// Try to create and then solve 3 unique problems using one or many of the array methods listed above.
+
+// Problem 1 want to return first name, last name and shirt size for record purpose; 
+const runnersInfoAndShirtsize = runners.filter((person) => {
```

Comment:
How about `map` method instead? "The `filter()` method creates a new array with all elements that pass the test implemented by the provided function." And I'm not seeing any test implemented inside `filter()`, therefore it returns the same array of objects that you had in the first place.

Reply:
https://javascript.info/array-methods